### PR TITLE
migrate cloud unit tests to pytest

### PR DESCRIPTION
**Pull request by MKLeb**
### What does this PR do?
migrates cloud unit tests to pytest

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by MKLeb** [comment]
re-run all

File: tests/pytests/unit/cloud/clouds/test_dimensiondata.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+try:
+    import libcloud.security
+
+    HAS_LIBCLOUD = True
+except ImportError:
+    HAS_LIBCLOUD = False
+
+
+VM_NAME = "winterfell"
```
Comment (garethgreenaway) [inline tests/pytests/unit/cloud/clouds/test_dimensiondata.py:25]: Was going to suggest that this should be a fixture, but I don't see where it's being used.

**Review by twangboy — DISMISSED** [review]

File: tests/unit/cloud/__init__.py [hunk 2]
```diff
@@ -1,66 +0,0 @@
...
-"""
-import pytest
-
-import salt.cloud
-from tests.support.unit import TestCase
-
-
-class CloudTest(TestCase):
-    @pytest.mark.slow_test
-    def test_vm_config_merger(self):
```
Comment (Ch3LL) [inline tests/unit/cloud/__init__.py:13]: I'm not seeing where this test was ported over to pytest

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by Ch3LL — DISMISSED** [review]

File: tests/unit/cloud/__init__.py (hunk 2 above)
Comment (MKLeb) [reply tests/unit/cloud/__init__.py:13]: done

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**